example/src/App.tsx

```diff
@@ -167,6 +171,11 @@ const App = () => {
     setBalancePerAddress(updatedBalances);
   };
 
+  const refreshSmartWalletAddresses = async () => {
+    const fetchedSmartWalletAddresses = await smartWalletAddresses().catch(() => []);
```

Comment:
would appreciate if this could go with try/catch instead of `await` and `.catch` because these 2 are different promise solving mixed together (either `.then().catch()` or t`ry { await } catch () {}`

Reply:
`.then()` is automatically called / returned if it's not in an error state and the .catch is an override so if the error occurs in the example folder, we can at least return an empty array... does that make sense? I tend to reserve try / catch for errors that cannot be caught with a `.catch` (as nothing can escape it)

Reply:
ngl not a fan of this as this pattern is later carried by others and sometimes mistaken for non-promise calls to catch the error so I'm always trying to keep try/catch in any scenario where error needs to be handled whether it's a promise or not

Reply:
Yeah okay understood - i'd rather go with a pattern that would stop others making mistakes going forward and try / catch is clearer. Will update 👍